src/backend/distributed/executor/multi_server_executor.c

```diff
@@ -104,12 +106,11 @@ JobExecutorType(DistributedPlan *distributedPlan)
 								errhint("Set citus.enable_repartition_joins to on "
 										"to enable repartitioning")));
 			}
-
-			ereport(DEBUG1, (errmsg(
-								 "cannot use adaptive executor with repartition jobs"),
-							 errhint("Since you enabled citus.enable_repartition_joins "
-									 "Citus chose to use task-tracker.")));
-			return MULTI_EXECUTOR_TASK_TRACKER;
+			if (ShardReplicationFactor > 1)
+			{
```

Comment:
This comparison uses the current value of the GUC, not the actual replication factor. What happens if the GUC changes to 1 after creating a table with replication factor 2?

Reply:
Then it would work incorrectly. We should use the replication factor at the creation time here.